packages/wallet/src/cip30.ts

```diff
@@ -290,21 +290,44 @@ export const createWalletApi = (
       data: coreTx,
       type: Cip30ConfirmationCallbackType.SignTx
     }).catch((error) => mapCallbackFailure(error, logger));
+
     if (shouldProceed) {
       const wallet = await firstValueFrom(wallet$);
       try {
         const {
-          witness: { signatures }
+          tx: {
+            witness: { signatures }
+          },
+          partialSign
         } = await wallet.finalizeTx({ tx: { ...coreTx, hash } });
 
+        // If partialSign is false and the wallet could not sign the entire transaction
+        if (!_partialSign && partialSign)
```

Comment:
Let's reserve the `partialSign` for the name of the argument (as it is in the cip), and rename our variable here.
```suggestion
        if (!partialSign && signedPartially)
```